### PR TITLE
Fallback to retry with HttpClient when RestSharp fails

### PR DESCRIPTION
**Pull request by IgalKreich**

### CODE REVIEW

File: Atlassian.Jira/Remote/JiraRestClient.cs [hunk 1]
```diff
@@ -258,46 +260,41 @@ private JToken GetValidJsonFromResponse(IRestRequest request, IRestResponse resp
...
+        private async Task<JToken> FallbackToHttpClientOnFailureAsync(IRestRequest request)
         {
-            try
+            using var handler = new HttpClientHandler
             {
-                using var handler = new HttpClientHandler()
-                {
-                    AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate
-                };
-                handler.ServerCertificateCustomValidationCallback += (sender, certificate, chain, errors) => true;
```
Comment (tomer-amir-apiiro) [inline Atlassian.Jira/Remote/JiraRestClient.cs:269]: why did you remove? aren't we ignoring SSL errors?

Comment (IgalKreich) [reply Atlassian.Jira/Remote/JiraRestClient.cs:269]: I moved it to the constructor... look at L268 in the new version

**Review by tomer-amir-apiiro — APPROVED** [review]

**Review by liiri — APPROVED** [review]
if http client works, why not just replace the impl with it?

**Comment by IgalKreich** [comment]
> if http client works, why not just replace the impl with it?

Let's see if it actually works.  If it will solve the problem, I'll make this the default implementation
